module-2/backend/week_two.md

```diff
@@ -25,30 +36,93 @@ end
 ```
 
 Now how would you find the owner of the team with an id of 4?
-
+  owner = Team.find_by(id: 4).owner
+  
 3. What do they allow you to do?
+
+  The belongs_to line established a relationship, for ?Active Record?, between the Team and the Owner, so it allows you to call
+  .owner on a team and ActiveRecord references the database for with the owner_id to return the Owner.
+  
 7. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+
+  It's going to be a many-to-many relationship; I'm not sure how to draw the schema diagram in this box, but I think with that relationship it could involve a table like Students_Teachers that holds rows of each student matched to each teacher based on foreign keys.
+  
+  
 8. Define foreign key, primary key, and schema.
+
+  A primary key is the unique ID of a row(which is an instance of a model/object).
+  A foreign key is a primary key that is dropped into another table with '_id' after it to allow for linking between those    instances.
+  A schema is the skeleton of your database that defines all of the columns in each with their names as well as the kind of information it is(string, integer, etc).  You never edit it directly.
+  
 9. Describe the relationship between a foreign key on one table and a primary key on another table.
+
+  If I'm understanding the question correctly, the two should be equal values.  The foreign key allows for reference to the row by looking for the primary key that matches the foreign key.
+  
 10. What are the parts of an HTTP response?
-11. Describe some techniques to make our Sinatra code more DRY. Give an example of when you would use these techniques.
 
+  The request is sent to the controller.  From there, the controller talks to the model, giving it parameters.  The model talks to the database to get what it needs.  This is returned to the controller which then passes that information to the assigned view which is rendered with that information.
```

Comment:
This is how the MVC works, but ultimately all of this is the process that's used to generate an HTTP response. The response itself has two parts: headers, and body. The status code is the first of the headers.